purchase_invoice_plan/models/purchase.py

```diff
@@ -210,6 +188,16 @@ class PurchaseInvoicePlan(models.Model):
         store=True,
     )
 
+    @api.depends("percent")
+    def _compute_amount(self):
+        for rec in self:
+            rec.amount = rec.percent * rec.purchase_id.amount_untaxed / 100
+
+    @api.onchange("amount")
+    def _inverse_amount(self):
+        for rec in self:
+            rec.percent = rec.amount / rec.purchase_id.amount_untaxed * 100
```

Comment:
if amount_untaxed = 0, This will error. Please add condition 

```
if rec.purchase_id.amount_untaxed != 0:
     rec.percent = rec.amount / rec.purchase_id.amount_untaxed * 100
```

Reply:
Fixed